shared/src/metabase/shared/util/internal/time.cljs

```diff
@@ -166,13 +166,19 @@
   (.diff after before "day"))
 
 (defn- matches-time? [input]
-  (re-matches #"\d\d:\d\d(?::\d\d(?:\.\d+)?)?" input))
+  (re-matches #"^\d\d:\d\d(?::\d\d(?:\.\d+)?)?$" input))
 
 (defn- matches-date? [input]
-  (re-matches #"\d\d\d\d-\d\d-\d\d" input))
+  (re-matches #"^\d\d\d\d-\d\d-\d\d$" input))
 
 (defn- matches-date-time? [input]
-  (re-matches #"\d\d\d\d-\d\d-\d\dT\d\d:\d\d(?::\d\d(?:\.\d+)?)?" input))
+  (re-matches #"^\d\d\d\d-\d\d-\d\dT\d\d:\d\d(?::\d\d(?:\.\d+)?)?(?:Z|-\d\d:\d\d)?$" input))
+
+(defn- coerce-local-date-time [input]
+  (let [offset (re-find #"(?:Z|-\d\d:\d\d)$" input)]
+   (cond-> input
+    offset (subs 0 (- (count input) (count offset)))
```

Comment:
It looks like this just strips the offset. Since we do a regex match anyway, this could be done simpler by `clojure.string/replace`.